### DISCUSSION TITLE
Serialize relationship in SQLModel by default

### DISCUSSION THREAD

**Discussion by JakNowy** [fastapi/sqlmodel#734] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class Search(SQLModel, table=True):
    id: int = Field(default=None, nullable=True, primary_key=True, index=True)
    videos: list["Video"] = Relationship(back_populates="search")


class Video(SQLModel, table=True):
    id: int = Field(default=None, nullable=True, primary_key=True, index=True)
    search_id: int = Field(foreign_key="search.id")
    search: "Search" = Relationship(back_populates="videos")


@router.get("/", response_model=Video)
def read_videos(
    db: Session = Depends(deps.get_db),
):
    return db.query(Video).first()
```

### Description

The code above works fine, but does not return nested dictonary with search relationship. If I simply create

```
class VideoResponse(SQLModel):
    id: int
    search_id: int
    search: Search
```

and change response_model to VideoResponse, everything works as I wish. However it brings some redundancy, which in fact SQLModel aims to remove. Would it be possible to make SQLModel serialize the relationship by default?

### Operating System

Linux

### SQLModel Version

sqlmodel==0.0.14

### Python Version

3.11.4

**Comment by wangxin688** [comment]
I guess not. explicity is always better than implicity. Also, it's not woring on async session. Use `joinload/subquery/selectinload` is much more performance way. if you want to load it direclty, may be you should add `lazyload` to relationship.

**Comment by eddie-lab** [comment]
Unsubscribe Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by YuriiMotov** [comment]
If SQLModel serialized relationships by default, you would have an infinite loop with the models you shared.

I guess it's probably possible to come up with solution how to serialize relationships but prevent infinite loops. If you have such ideas - feel free to share!
